presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -498,6 +498,13 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                 }
             }
 
+            if (node.getDistributionType().isPresent()
```

Comment:
it seems that this check is unnecessary really. It's sub-optimal to use partitioned distribution without any equi-clause, but the execution shouldn't fail (partitioning on empty symbols is constant).

Maybe we should remove the check here? We could have a plan sanity checker instead or generate warning in such case.